### ISSUE TITLE
Class names aren't processed on plugin

### ISSUE THREAD

**Issue by BastiaanOlij** [godotengine/godot#85467]
### Godot version

4.1.3-stable

### System information

Windows 10

### Issue description

Due to dependency issues, `global_script_class_cache.cfg` isn't being populated after installing plugin.
This is a regression issue in 4.1.3 (and probably 4.1.2 and 4.1.1) but already fixed in 4.2.

### Steps to reproduce

This is easiest to reproduce with the TiltFive plugin, but also happens to other plugins such as XR Tools. You do not need a tilt five to reproduce the issue, but you do need Windows.

1. Create a new project
2. Open the asset library and search for TiltFive
3. Install the TiltFive plugin (Godot may crash, just reopen)
4. Errors should already be visible but they will disappear on restart (but aren't really gone)
5. Enable the TiltFive plugin, it's accessing classes again and thus new errors
6. Create a new scene and make this your startup scene
7. Add a Node3D to the scene, assign the `res://addons/tiltfive/T5Manager.gd` script to this node
8. Save your scene

Every time you load up the scene these errors are reported:
![image](https://github.com/godotengine/godot/assets/1945449/951a9098-7b2c-41d8-97c6-fd2f5076f496)

When you open up one of the effected scripts such as T5ProjectSettings you'll see that the class_name remains unresolved:
![image](https://github.com/godotengine/godot/assets/1945449/473b01c9-8155-4617-b38d-bcf59341ac52)

Opening up `global_script_class_cache.cfg` you will see none of the class names are added:
```
list=[]
```

The workaround is to edit each script that has `class_name` and rename the class, and then rename it back so it gets added. 

### Minimal reproduction project

n/a

**Comment by BastiaanOlij** [comment]
Note also that the same issues are present when cloning an existing project that uses the tiltfive plugin, or when you delete the `.godot` folder to start from scratch (easy way to reproduce this on a working project).

Here is a full project that can be cloned and opened in Godot: https://github.com/GodotVR/TiltFiveTools

**Comment by YuriSizov** [comment]
> This is a regression issue in 4.1.3 (and probably 4.1.2 and 4.1.1) but already fixed in 4.2.

If it's already present in 4.1.x then there is no point marking it as a regression, it's just a bug introduced some time ago. 🙂

**Comment by BastiaanOlij** [comment]
Yup, definately worked in 4.0, and as far as I'm aware in 4.1, broke sometime later, but seems to be fixed in 4.2

**Comment by Malcolmnixon** [comment]
We're getting reports of this issue occurring in 4.2-stable, but I'm unable to reproduce:
![image](https://github.com/godotengine/godot/assets/1863707/1c56a766-ec97-4f87-96ab-659c57a375e8)

**Comment by BastiaanOlij** [comment]
@YuriSizov @akien-mga now that the Godot XR game jam has started I'm getting reports from people that this issue does exist in Godot 4.2 as well, so either we reverted something that fixed it, or there is still a condition where this happens.

I'm going to try and reproduce it myself to verify the claims, but it has been reported with the TiltFive plugin.

One theory I have, but I could be barking up the wrong tree here, is that I wonder if when a plugin is disabled, we stop the scripts in that plugin from being evaluated. As for TiltFive you actually install two plugins that have a dependency, and we can't enable the plugin if the plugin script can't be run, but it can't be run without the classes being registered, and we got ourselves a catch-22 situation.

**Comment by BastiaanOlij** [comment]
@Malcolmnixon what stands out in that error log are the errors on classes `T5Origin3D` and `T5Camera3D` which are all classes implemented in gdextension. This feels like the extension not being loaded.

**Comment by Malcolmnixon** [comment]
> @Malcolmnixon what stands out in that error log are the errors on classes `T5Origin3D` and `T5Camera3D` which are all classes implemented in gdextension. This feels like the extension not being loaded.

I was thinking the same, but then they responded with:
```
threw a couple of scripts into autoload (Plugin.gd and T5Interface.gd as the second didn't fix the problem alone) and it still throws errors but now I at least have it to where I finally have it generating my first holographic cube! (kinda neat!)
Now to get back to the meat of what I need to start on for now
```

This suggests they're cognizant about the plugin enabling, and there's enough running that they can get the headset to work; so I'm still stumped.
